src/main/update/check.ts

```diff
@@ -0,0 +1,102 @@
+import { IApv, ISimpleApv } from "src/interfaces/apv";
+import { IDownloadUrls, getDownloadUrls } from "../../utils/url";
+import Headless from "../headless/headless";
+import { get as getConfig, baseUrl, netenv } from "../../config";
+
+export class GetPeersApvFailedError extends Error {}
+
+export interface IUpdate {
+  newApv: ISimpleApv;
+  oldApv: ISimpleApv;
+  urls: IDownloadUrls;
+}
+
+const peerInfos = getConfig("PeerStrings");
+const localApvToken = getConfig("AppProtocolVersion");
+const trustedApvSigners = getConfig("TrustedAppProtocolVersionSigners");
+
+export async function checkForUpdate(
+  standalone: Headless,
+  platform: NodeJS.Platform
+): Promise<IUpdate | null> {
+  let peersApv;
+  try {
+    peersApv = getPeersApv(standalone, peerInfos, trustedApvSigners);
+  } catch (e) {
+    console.error(`getPeersApv Error ocurred ${e}:\n`, e.stderr);
+    return null;
+  }
+
+  const localApv = standalone.apv.analyze(localApvToken);
+
+  if (peersApv.version > localApv.version)
+    return {
+      newApv: peersApv,
+      oldApv: localApv,
+      urls: getDownloadUrls(baseUrl, netenv, peersApv.version, platform),
+    };
+
+  return null;
+}
+
+// FIXME: Could use overload function...
+export async function checkForUpdateUsedPeersApv(
+  standalone: Headless,
+  peersApv: ISimpleApv,
+  platform: NodeJS.Platform
+): Promise<IUpdate | null> {
+  const localApv = standalone.apv.analyze(localApvToken);
+
+  if (peersApv.version > localApv.version)
+    return {
+      newApv: peersApv,
+      oldApv: localApv,
+      urls: getDownloadUrls(baseUrl, netenv, peersApv.version, platform),
+    };
+
+  return null;
+}
+
+/**
+ * Checks `CompatiblityVersion` to check if we can proceed with our updater.
+ */
+export function checkCompatiblity(
+  peersApv: ISimpleApv,
+  localApv: ISimpleApv
+): boolean {
+  const peersCompatVersion = BigInt(
+    (peersApv.extra["CompatiblityVersion"] as string | number) ?? 0
+  );
+  const localCompatVersion = BigInt(
+    (localApv.extra["CompatiblityVersion"] as string | number) ?? 0
+  );
+
+  if (peersCompatVersion > localCompatVersion) return false;
+
+  return true;
```

Comment:
```suggestion
  return peersCompatVersion <= localCompatVersion;
```

Simplifying things a bit.

Reply:
Wow Genius

Reply:
Done